addons/binding/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/handler/VelbusBridgeHandler.java

```diff
@@ -0,0 +1,172 @@
+                    logger.debug("Failed to register event listener on serial port {}", getConfig().get(PORT));
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to connect to serial port " + getConfig().get(PORT));
+                logger.debug("Failed to connect to serial port {}", getConfig().get(PORT));
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Serial port name not configured");
+            logger.debug("Serial port name not configured");
+        }
+    }
+
+    public synchronized void sendPacket(byte[] packet) {
+        delaySendPacket();
+
+        try {
+            outputStream.write(packet);
+            outputStream.flush();
+        } catch (IOException e) {
+            logger.error("Serial port write error", e);
+        }
+
+        lastPacketTimeMillis = System.currentTimeMillis();
+    }
+
+    /*
+     * When sending you need a delay of 60ms between each packet
+     * (to prevent flooding the VMB1USB).
+     */
+    protected void delaySendPacket() {
+        long currentTimeMillis = System.currentTimeMillis();
+        long timeSinceLastPacket = currentTimeMillis - lastPacketTimeMillis;
+
+        if (timeSinceLastPacket < 60) {
+            long timeToSleep = 60 - timeSinceLastPacket;
+
+            try {
+                Thread.sleep(timeToSleep);
+            } catch (InterruptedException e) {
+                logger.error("The packet delay was interrupted.", e);
+            }
+        }
+    }
+
+    public void setDefaultPacketListener(VelbusPacketListener velbusPacketListener) {
+        defaultPacketListener = velbusPacketListener;
+    }
+
+    public void registerPacketListener(byte address, VelbusPacketListener packetListener) {
+        if (packetListener == null) {
+            throw new IllegalArgumentException("It's not allowed to pass a null RelayStatusListener.");
+        }
+
+        packetListeners.put(Byte.valueOf(address), packetListener);
+    }
+
+    public void unregisterRelayStatusListener(byte address, VelbusPacketListener packetListener) {
+        packetListeners.remove(Byte.valueOf(address));
+    }
+
+    @Override
+    public void dispose() {
+        if (serialPort != null) {
+            serialPort.disconnect();
+            serialPort = null;
+        }
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+        logger.debug("Serial port event triggered");
+
+        if (arg0.getEventType() == SerialPortEvent.DATA_AVAILABLE) {
+            try {
+                byte[] packet;
+                while ((packet = inputStream.readPacket()) != null) {
+                    byte address = packet[2];
+
+                    VelbusPacketListener packetListener = packetListeners.get(address);
```

Comment:
The packetListeners map contains the link between a device address and a handler (=packet listener). Since an address uniquely identifies a device, and there is only one handler per device, it should not be possible to have multiple listeners for one address.